### ISSUE TITLE
Improve optimizing apps experience after update

### ISSUE THREAD

**Issue by chriswoope** [GrapheneOS/os-issue-tracker#2082]
After updates, GrapheneOS appears to compile all apps ahead of time, which can take an hour or more if hundreds or thousands of apps are installed. This can be mitigated by rebooting on idle or after going to sleep, but sometimes one might still need to use the device at that time.

First of all, it's not clear why the whole system needs to be blocked by this, since it seems like the compilation could be done in the background at least for non-system apps (ideally prioritizing by last app usage time), and the blocking could happen only when an app that is not yet compiled gets started.

If the blocking is not changed, then it would be nice to have at least a clock, battery charge level and elapsed/estimated time shown on the screen, and ideally also working alarms, a flashlight button and if feasible a way to call emergency numbers or even an unlocked SIM and a full dialer.

This is to some extent an upstream issue as well, but is particularly problematic for GrapheneOS due to having very frequent updates and completely disabling JIT compilation of apps.

**Comment by thestinger** [comment]
It's meant to recompile the apps before rebooting during the Finalizing step. If that's not working properly for all apps, that's what needs to be addressed.

**Comment by thestinger** [comment]
@chriswoope I'm aware if you spreading misinformation about GrapheneOS elsewhere and you aren't welcome here. You're banned from our community including on GitHub going forward. Escalate further with more misinformation and it will only get worse for you.
